pybaselines/_algorithm_setup.py

```diff
@@ -32,37 +34,30 @@ def difference_matrix(data_size, diff_order=2):
     Raises
     ------
     ValueError
-        Raised if diff_order is not 0, 1, 2, 3, 4, or 5.
+        Raised if diff_order is not positive.
 
     Notes
     -----
     Most baseline algorithms use 2nd order differential matrices when
     doing penalized least squared fitting.
 
-    It would be possible to support any differential order by doing
-    np.diff(np.eye(data_size), diff_order), but the resulting matrix could
-    cause issues if data_size is large. Therefore, it's better to only
-    provide sparse arrays for the most commonly used differential orders.
-
     The resulting matrices are transposes of the result of
-    np.diff(np.eye(data_size), diff_order). Not sure why there is a discrepancy,
-    but this implementation allows using the differential matrices are they
-    are written in various publications, ie. D.T * D rather than having to
-    do D * D.T.
+    np.diff(np.eye(data_size), diff_order). This implementation allows using
+    the differential matrices are they are written in various publications,
+    ie. D.T * D rather than having to do D * D.T.
 
     """
-    if diff_order not in (0, 1, 2, 3, 4, 5):
-        raise ValueError('The differential order must be 0, 1, 2, 3, 4, or 5')
-    diagonals = {
-        0: [1],
-        1: [-1, 1],
-        2: [1, -2, 1],
-        3: [-1, 3, -3, 1],
-        4: [1, -4, 6, -4, 1],
-        5: [-1, 5, -10, 10, -5, 1]
-    }[diff_order]
+    if diff_order < 0:
+        raise ValueError('The differential order must be positive')
+    if diff_order > data_size:
+        diff_order = data_size
+
+    diagonals = np.zeros(2 * diff_order + 1)
+    diagonals[diff_order] = 1
+    for _ in repeat(None, diff_order):
```

Comment:
Can range be used here instead of repeat since it is just used for iterating?

Reply:
In this case, since most of the time `diff_order` will be a fairly small value. Use of `range` won't make much of a difference. So I'm swapping it back to `range`.
However, if it's just for iterating, you are better off using `repeat(None, x)` than `range(x)` performance-wise. I was planning to make another PR at some point for the different algorithms.